spec/worker_spec.rb

```diff
@@ -45,6 +45,49 @@
         expect(job).to_not be_nil
         expect(job['last_error']).to_not be_nil
       end
+
+      context 'for a non-QueueClassicPlus job' do
+        after { Object.send(:remove_const, :NonQcJob) }
+
+        let(:job_type) do
+          Class.new(ActiveJob::Base) do
+            def self.name
+              "NonQcJob"
+            end
+
+            self.queue_adapter = :queue_classic
+
+            queue_as :test
+
+            def perform(should_raise)
+              raise "boom" if should_raise
```

Comment:
Is the conditional needed?

Reply:
It's not, this is cargo-culted from the other fake job classes used in specs (not all of which need the conditional).